Sources/FoundationEssentials/Data/Data.swift

```diff
@@ -2710,13 +2710,34 @@ public struct Data : Equatable, Hashable, RandomAccessCollection, MutableCollect
     @inlinable // This is @inlinable as emission into clients is safe -- the concept of equality on Data will not change.
     public static func ==(d1 : Data, d2 : Data) -> Bool {
         let length1 = d1.count
-        if length1 != d2.count {
+        let length2 = d2.count
+        
+        // Unequal length data can never be equal
+        guard length1 == length2 else {
             return false
         }
+
+        // See if both are empty
+        switch (d1._representation, d2._representation) {
```

Comment:
Does the `if length1 > 0` check below cover this case or is there a possibility that we'd somehow end up with an `.empty` representation with a length of greater than 0?

Reply:
I think an empty representation is always empty, right? I suppose we could even move this check above the count if that clarifies things.